src/sagemaker_containers/encoders.py

```diff
@@ -38,17 +38,36 @@ def array_to_npy(array_like):  # type: (np.array or Iterable or int or float) ->
     return buffer.getvalue()
 
 
-def npy_to_numpy(npy_array):  # type: (object) -> np.array
+def npy_to_numpy(npy_array, dtype=np.float32):  # type: (object) -> np.array
     """Convert an NPY array into numpy.
 
     Args:
+        dtype (dtype, optional):  Data type of the resulting array.
+                                    If None, the dtypes will be determined by the contents of each column, individually.
         npy_array (npy array): to be converted to numpy array
 
     Returns:
         (np.array): converted numpy array.
     """
     stream = BytesIO(npy_array)
-    return np.load(stream)
+    return np.load(stream).astype(dtype)
+
+
+def unicode_to_str(string_like):  # type: (str or unicode) -> str
+    """Convert unicode in string. If it is array a string, returns itself.
+
+    Args:
+        string_like (str or unicode): string like object to be converted to string
+
+    Returns:
+        (str): converted string
+    """
+    if hasattr(string_like, 'decode'):
+        try:
+            return string_like.decode('utf-8')
+        finally:
+            return string_like.decode('latin1')
```

Comment:
This will always return `string_like.decode('latin1')`. I think this should be try: return utf-8, catch some exception, return latin1.